### PR TITLE
[TASK] Add missing test case for fetching none existing tea

### PR DESCRIPTION
**Pull request by DanielSiepmann**
Dedicated test cases were added in the past.
This test case was not added yet.

Resolves: #858

### CODE REVIEW

File: Tests/Functional/Domain/Repository/Product/TeaRepositoryTest.php [hunk 1]
```diff
@@ -66,6 +66,15 @@ public function findAllSortsByTitleInAscendingOrder(): void
         self::assertSame(2, $result->current()->getUid());
     }
 
+    /**
+     * @test
+     */
+    public function findByUidForNoneExistingRecord(): void
```
Comment (oliverklee) [inline Tests/Functional/Domain/Repository/Product/TeaRepositoryTest.php:72]: Let's make the test name a full sentence containing a statement about the checked behavior:

```suggestion
    public function findByUidForInexistentRecordReturnsNull(): void
```

https://speakerdeck.com/oliverklee/test-driven-development-with-phpunit-915a5f2c-3d37-4e41-b5bc-ef1efca9c01e?slide=39

File: Tests/Functional/Domain/Repository/Product/TeaRepositoryTest.php [hunk 2]
```diff
@@ -66,6 +66,15 @@ public function findAllSortsByTitleInAscendingOrder(): void
...
+    {
+        $model = $this->subject->findByUid(1);
```
Comment (oliverklee) [inline Tests/Functional/Domain/Repository/Product/TeaRepositoryTest.php:74]: Please let's add a blank line to visually separate the "execute" step from the "verify" step.

https://speakerdeck.com/oliverklee/test-driven-development-with-phpunit-915a5f2c-3d37-4e41-b5bc-ef1efca9c01e?slide=22

**Review by oliverklee — CHANGES_REQUESTED** [review]

**Review by oliverklee — APPROVED** [review]
